### ISSUE TITLE
panic: ./kube-monkey flag redefined: log_dir

### ISSUE THREAD

**Issue by Joeskyyy** [asobti/kube-monkey#37]
Hey there!

I've tried following the instructions explicitly on the README, but unfortunately I keep hitting an interesting blocked, that I *think* may be tied to some circular dependencies in glog (I'm a super novice at Go, admittedly).

In my error log I get the following:

```
/kube-monkey flag redefined: log_dir
panic: /kube-monkey flag redefined: log_dir

goroutine 1 [running]:
flag.(*FlagSet).Var(0xc42004c060, 0x1d8fe00, 0xc42000efa0, 0x14c1722, 0x7, 0x14e3d51, 0x2f)
	/usr/local/Cellar/go/1.9.3/libexec/src/flag/flag.go:793 +0x5e1
flag.(*FlagSet).StringVar(0xc42004c060, 0xc42000efa0, 0x14c1722, 0x7, 0x0, 0x0, 0x14e3d51, 0x2f)
	/usr/local/Cellar/go/1.9.3/libexec/src/flag/flag.go:696 +0x8b
flag.(*FlagSet).String(0xc42004c060, 0x14c1722, 0x7, 0x0, 0x0, 0x14e3d51, 0x2f, 0xc4200f1f00)
	/usr/local/Cellar/go/1.9.3/libexec/src/flag/flag.go:709 +0x8b
flag.String(0x14c1722, 0x7, 0x0, 0x0, 0x14e3d51, 0x2f, 0x0)
	/usr/local/Cellar/go/1.9.3/libexec/src/flag/flag.go:716 +0x69
github.com/asobti/kube-monkey/vendor/k8s.io/client-go/1.5/vendor/github.com/golang/glog.init()
	/Users/joe/go/src/github.com/asobti/kube-monkey/vendor/k8s.io/client-go/1.5/vendor/github.com/golang/glog/glog_file.go:41 +0x14a
github.com/asobti/kube-monkey/vendor/k8s.io/client-go/1.5/kubernetes.init()
	<autogenerated>:1 +0x48
github.com/asobti/kube-monkey/deployments.init()
	<autogenerated>:1 +0x53
github.com/asobti/kube-monkey/chaos.init()
	<autogenerated>:1 +0x5d
github.com/asobti/kube-monkey/kubemonkey.init()
	<autogenerated>:1 +0x53
main.init()
	<autogenerated>:1 +0x5d
```

This is before even trying to run it in k8s, and just running it locally, but we saw the exact same issues when running this in k8s.

I've tried some attempts at resolving conflicts using `glide install --strip-vendor` but to no avail.

Just wondering if you have any words of advice, or something stupid I may be doing as a Go noob (:  

Additionally, do you think it's possible to push your image as a release to hub.docker.com?

**Comment by asobti** [comment]
Interesting. I actually had this issue the first time I tried to implement logging using `glog`, which is why I had abandoned the attempt then.

@Spellchaser any ideas?

**Comment by Spellchaser** [comment]
Glide doesn't seem to update properly and since master still uses client-go 1.5, there's dependencies that break if you try to update properly. If you were to build the dependencies from scratch -
```
glide cc
mv glide.yaml glide.bak
rm glide.lock
rm -rf vendor/
glide init 
glide up -v --all-dependencies 
```
It'll break, even if you set version of client-go to 1.5 and do all sorts of glide trickery. 

Verified that master isn't building at the moment. 
I think your glide update didn't strip vendor and there isn't an easy way to fix with glide... There might be a hacky fix by installing glog locally...

**Comment by Spellchaser** [comment]
tl;dr glide sucks, client-go version is too old

I don't think it's productive to try to fix the glide issue since my client-go update PRs fix this. I'm not sure when master broke but it must have been before we started messing with glide. 

I know a former commit of master worked and I just verified my fork builds and works from a complete scratch vm.

**Comment by Joeskyyy** [comment]
@Spellchaser 👍 I tried to build your fork actually earlier haha, I was having much of the same issue, let me try again to verify for you with a clean slate.

**Comment by Joeskyyy** [comment]
Confirmed it works in your fork now @Spellchaser with your new directions (thank you very much for that!). Small missing piece, need to `cd` into `kube-monkey.git` after the clone (: 

Thanks to you both for the super fast responses! Gonna close this in favour of awaiting changes, for now I'll build from the fork.

Would still be awesome to have verified, and shipped images to docker hub if you ever get the chance!

**Comment by Spellchaser** [comment]
Ahh yes... it's totally omitted to avoid straight copy pasta ;)

**Comment by asobti** [comment]
@Joeskyyy Agreed on the docker hub images. 

Once I work through all of @Spellchaser's PRs, I'll build and publish an image
